chain/chain/src/chain.rs

```diff
@@ -3109,13 +3225,10 @@ impl<'a> ChainUpdate<'a> {
         prev_block_hash: CryptoHash,
     ) -> Result<(), Error> {
         match result {
-            ApplyChunkResult::SameHeight(SameHeightResult {
-                shard_id,
-                gas_limit,
-                apply_result,
-            }) => {
+            ApplyChunkResult::SameHeight(SameHeightResult { gas_limit, apply_result }) => {
                 let (outcome_root, outcome_paths) =
                     ApplyTransactionResult::compute_outcomes_proof(&apply_result.outcomes);
+                let shard_id = apply_result.trie_changes.shard_uid.shard_id();
```

Comment:
this way of getting shard id is a bit strange. What is the reason for this?

Reply:
I agree it looks a bit hacky, I changed it